### PR TITLE
Finishing tasks

### LINKED ISSUE #14: finishing tasks (child)
lets give our children the ability to finish tasks

### Value Proposition

As a user i want to finish tasks to be able to gain rewards for it

### Description

Screenshot 2022-12-12 at 11 13 02

### Acceptance Criteria

- [ ] when logged in as a child, you see a "done" button
- [ ] when clicking on that button, you get a popup which asks you if you are sure to finish the task
- [ ] after confirming the task will disappear from the list and you get a message, that it will be reviewed by the parent
- [ ] when logged in as parent you can confirm the completed task

### Tasks

(for developers team)

- [ ] add a check button linked to "child" state
- [ ] add a popup for the button
- [ ] add "review" key for tasks
- [ ] update the value for "review" to "in review" when clicked on "confirm"
- [ ] filter out "in review" tasks (homepage, children)
- [ ] (parent) filter tasks, with "in review" tasks shown at the top of the page
- [ ] add a "confirm" button to update the "in review" value to "reviewed"

### Sizing/Complexity

**M**

### PR DESCRIPTION
**Pull request by MaxMer25**
Children can now finish tasks and mark them for review for parents.
parents can confirm the review

### CODE REVIEW

File: pages/home/addTask.js [hunk 1]
```diff
@@ -39,12 +42,16 @@ export default function AddTask() {
...
-  const resetPopup = () => {
-    setPopup(false);
-  };
+  function triggerPopup() {
+    if (popup == true) {
+      setPopup(false);
+    } else {
+      setPopup(true);
+    }
+  }
```
Comment (stefankorder) [inline pages/home/addTask.js]: I think you don't need this function. Take a look at my comment above where you call this function.

File: pages/home/addTask.js [hunk 2]
```diff
@@ -39,12 +42,16 @@ export default function AddTask() {
   async function handleSubmit(event) {
     event.preventDefault();
     await submitTask();
-    setPopup(!popup);
+    triggerPopup();
```
Comment (stefankorder) [inline pages/home/addTask.js:45]: You could just write:

```suggestion
   setPopup(previousState => !previousState)
```

**Review by stefankorder — CHANGES_REQUESTED** [review]
Good work! Take a look at my comments.

File: pages/home/addTask.js (hunk 2 above)
Comment (MaxMer25) [reply pages/home/addTask.js:45]: thanks for your suggestion, i fixed it 👍

**Pull request merged**